File: src/main/java/ru/mrgrd56/api/storage/storages/MappedStorage.java
package ru.mrgrd56.api.storage.storages;

public interface MappedStorage<T> {
    boolean has(String key);
    T get(String key);
    T set(String key, T value);
    T add(String key, T value);
    T update(String key, T value);
    void delete(String key);
}

File: src/main/kotlin/ru/mrgrd56/api/storages/MappedStorage.kt
package ru.mrgrd56.api.storages

interface MappedStorage<T> {
    fun has(key: String): Boolean
    operator fun get(key: String): T
    operator fun set(key: String, value: T): T
    fun add(key: String, value: T): T
    fun update(key: String, value: T): T
    fun delete(key: String)
}
